src/extension/csv-with-metadata-header/src/main/java/au/org/emii/geoserver/wfs/response/CSVWithMetadataHeaderOutputFormat.java

```diff
@@ -26,51 +14,168 @@
+        BufferedWriter w = new BufferedWriter(new OutputStreamWriter(output, this.gs.getGlobal().getSettings().getCharset()));
+        FeatureCollection<?, ?> fc = (FeatureCollection)featureCollection.getFeature().get(0);
+        Object att;
+
+        SimpleFeatureType ft = (SimpleFeatureType)((FeatureCollection)fc).getSchema();
+        w.write("FID,");
+
+        for(int i = 0; i < ft.getAttributeCount(); ++i) {
+            AttributeDescriptor ad = ft.getDescriptor(i);
+            w.write(this.prepCSVField(ad.getLocalName()));
+            if (i < ft.getAttributeCount() - 1) {
+                w.write(",");
+            }
+        }
+
+        w.write("\r\n");
+        NumberFormat coordFormatter = NumberFormat.getInstance(Locale.US);
+        coordFormatter.setMaximumFractionDigits(this.getInfo().getGeoServer().getSettings().getNumDecimals());
+        coordFormatter.setGroupingUsed(false);
+        FeatureIterator i = ((FeatureCollection)fc).features();
+
+        try {
+            for(; i.hasNext(); w.write("\r\n")) {
+                Feature f = i.next();
+                w.write(this.prepCSVField(f.getIdentifier().getID()));
+                w.write(",");
+
+                for(int j = 0; j < ((SimpleFeature)f).getAttributeCount(); ++j) {
+                    att = ((SimpleFeature)f).getAttribute(j);
+                    if (att != null) {
+                        String value = formatToString(att, coordFormatter);
+                        w.write(this.prepCSVField(value));
+                    }
+
+                    if (j < ((SimpleFeature)f).getAttributeCount() - 1) {
+                        w.write(",");
+                    }
+                }
+            }
+        } finally {
+            i.close();
+        }
+        w.flush();
+    }
+
+    private String prepCSVField(String field) {
+        String mod = field.replaceAll("\"", "\"\"");
+        if (CSV_ESCAPES.matcher(mod).find()) {
+            mod = "\"" + mod + "\"";
+        }
+        return mod;
+    }
+
+    public String formatToString(Object att, NumberFormat coordFormatter) {
+        String value = null;
+
+        if (att instanceof Number) {
+            value = coordFormatter.format(att);
+        } else if (att instanceof Date) {
+            if (att instanceof java.sql.Date) {
+                value = utcDateString((Date)att);
+            } else if (att instanceof Time) {
+                value = DateUtil.serializeSqlTime((Time)att);
+            }else {
+                value = utcDateString((Date)att);
+            }
+        } else {
+            value = att.toString();
+        }
+        return value;
+    }
+
+    private String utcDateString(Date date) {
+
+        try {
+            String dateUTC = date.toString();
+            DateTime dt = IMOS_DATE_TIME_FORMATTER.parseDateTime(dateUTC);
+
+            DateTimeFormatter fmt = ISODateTimeFormat.dateTimeNoMillis();
```

Comment:
Not sure why we need to convert to a string and then parse the string.  Why not just create the datetime from the date as follows:

    DateTime dt = new DateTime( date, DateTimeZone.UTC );

Reply:
I tried `DateTime dt = new DateTime( date, DateTimeZone.UTC );` and it doesn't work.

This essentially is the problem i've spent all the time on, I would need help on this to make a change beyond what i have currently